### PR TITLE
PT-5474: HttpClientFactory is registered and used instead webClient

### PR DESCRIPTION
**Pull request by getmansky**
## Description
WebClient is replaced with HttpClientFactory
### Jira-link:

https://virtocommerce.atlassian.net/browse/PT-5474

### CODE REVIEW

File: src/VirtoCommerce.Platform.Web/Controllers/Api/PlatformExportImportController.cs [hunk 1]
```diff
@@ -42,29 +44,30 @@ public class PlatformExportImportController : Controller
             IPushNotificationManager pushNotifier,
             ISettingsManager settingManager,
             IUserNameResolver userNameResolver,
-            IOptions<PlatformOptions> options)
+            IOptions<PlatformOptions> options, IHttpClientFactory httpClientFactory)
```
Comment (ksavosteev) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/PlatformExportImportController.cs:47]: Move to new line

**Review by ksavosteev — APPROVED** [review]

**Comment by mvktsk** [comment]
Review task https://virtocommerce.atlassian.net/browse/VP-7886 has been created

File: src/VirtoCommerce.Platform.Web/Controllers/Api/PlatformExportImportController.cs [hunk 2]
```diff
@@ -205,7 +208,7 @@ public ActionResult DownloadExportFile([FromRoute] string fileName)
             }
         }
 
-        private IEnumerable<SampleDataInfo> InnerDiscoverSampleData()
+        private async Task<IEnumerable<SampleDataInfo>> InnerDiscoverSampleData()
```
Comment (pushnitsa) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/PlatformExportImportController.cs:211]: InnerDiscoverSampleDataAsync()

File: src/VirtoCommerce.Platform.Web/Controllers/Api/PlatformExportImportController.cs [hunk 3]
```diff
@@ -42,32 +44,34 @@ public class PlatformExportImportController : Controller
...
         [Authorize(Permissions.PlatformImport)]
-        public ActionResult<SampleDataImportPushNotification> TryToAutoInstallSampleData()
+        public async Task<ActionResult<SampleDataImportPushNotification>> TryToAutoInstallSampleData()
         {
-            var sampleData = InnerDiscoverSampleData().FirstOrDefault(x => !x.Url.IsNullOrEmpty());
+            var sampleData = (await InnerDiscoverSampleData()).FirstOrDefault(x => !x.Url.IsNullOrEmpty());
             if (sampleData != null)
             {
-                return ImportSampleData(sampleData.Url);
+                return ImportSampleDataAsync(sampleData.Url);
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/PlatformExportImportController.cs:74]: Missing await?

File: src/VirtoCommerce.Platform.Web/Controllers/Api/PlatformExportImportController.cs [hunk 4]
```diff
@@ -224,8 +228,8 @@ private IEnumerable<SampleDataInfo> InnerDiscoverSampleData()
 
             //Discovery mode
             var manifestUrl = sampleDataUrl + "\\manifest.json";
-            using (var client = new WebClient())
-            using (var stream = client.OpenRead(new Uri(manifestUrl)))
+            using (var client = _httpClientFactory.CreateClient())
+            using (var stream = await client.GetStreamAsync(new Uri(manifestUrl)))
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/PlatformExportImportController.cs:232]:
```suggestion
            using var client = _httpClientFactory.CreateClient();
            using var stream = await client.GetStreamAsync(new Uri(manifestUrl));
```

**Review by Vectorfield4 — APPROVED** [review]

**Pull request merged**